src/utils/bitcoin.ts

```diff
@@ -174,7 +174,34 @@ export async function getTxProof(
             outputs: tx.outs.map(parseTxOutput),
             lockAt: parseLockTime(tx.locktime),
         },
-        lengthBound: tx.byteLength(),
+        txEncodedLen: tx.byteLength(),
+    };
+}
+
+export async function getTxProof(
+    electrsAPI: ElectrsAPI,
+    userTxId: string
+): Promise<{
+    userTxProof: PartialTxProof;
+    coinbaseProof: PartialTxProof;
+}> {
+    const coinbaseTxId = await electrsAPI.getCoinbaseTxId(userTxId);
+
+    if (coinbaseTxId === undefined) {
+        throw new Error(`Coinbase txid not found for transaction ${userTxId}`);
```

Comment:
Just for clarification: Is this going to be a problem if I send my BTC from a non-coinbase wallet?

Reply:
Coinbase wallet as the wallet developed by that CEX company? No, actually each bitcoin block has[ coinbase transaction ](https://developer.bitcoin.org/reference/transactions.html#coinbase-input-the-input-of-the-first-transaction-in-a-block)which is the first tx in the block with reward for the miner.

Reply:
Riiight, the name keeps confusing me :)